Build fails with `-DUSE_OPENMP=OFF`

### Is there an existing issue for this?

- [x] I checked and did not find my issue in the already reported ones

### Describe the bug

Building darktable-5.2 with `-DUSE_OPENMP=OFF` fails with
```
/builddir/darktable-5.2.0/src/common/darktable.c:1190:66: error: implicit declaration of function 'omp_get_dynamic' [-Wimplicit-function-declaration]
 1190 |           darktable.num_openmp_threads, (int)dt_get_num_procs(), omp_get_dynamic() ? "(dynamic)" : "(static)");
```
https://github.com/darktable-org/darktable/blame/f2e4da79af32180e9d6b3a48be30d4ba52ef8b6e/src/common/darktable.c#L1190

I think the build failure is due to the changes in #18903. The call to `omp_get_dynamic` aren't guarded by an `#ifdef _OPENMP`.


### Steps to reproduce

build with `-DUSE_OPENMP=OFF`

### Expected behavior

build :)

### Logfile | Screenshot | Screencast

_No response_

### Commit

48486b7fc6bd1472ec75e4b4d5bd3eb977bace5f

### Where did you obtain darktable from?

darktable.org / GitHub release

### darktable version

5.2

### What OS are you using?

Linux

### What is the version of your OS?

voidlinux

### Describe your system

_No response_

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

--- a/src/common/darktable.c
+++ b/src/common/darktable.c
@@ -1193,10 +1193,13 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
         if(desired > possible)
           dt_print(DT_DEBUG_ALWAYS,
                    "[dt_init --threads] requested %d ompthreads restricted to %d",
-            desired, possible);
+                   desired, possible);
+#ifdef _OPENMP
         dt_print(DT_DEBUG_ALWAYS,
                  "[dt_init --threads] using %d threads of %d for openmp parallel sections %s",
-          darktable.num_openmp_threads, (int)dt_get_num_procs(), omp_get_dynamic() ? "(dynamic)" : "(static)");
+                 darktable.num_openmp_threads, (int)dt_get_num_procs(),
+                 omp_get_dynamic() ? "(dynamic)" : "(static)");
+#endif
         k++;
         argv[k-1] = NULL;
         argv[k] = NULL;

